service/awsconfig/drainer_framework.go

```diff
@@ -1,16 +1,164 @@
 package awsconfig
 
 import (
+	"time"
+
+	"github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
+	"github.com/giantswarm/apiextensions/pkg/clientset/versioned"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/giantswarm/operatorkit/client/k8scrdclient"
 	"github.com/giantswarm/operatorkit/framework"
+	"github.com/giantswarm/operatorkit/informer"
+	apiextensionsclient "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	"k8s.io/client-go/kubernetes"
+
+	awsclient "github.com/giantswarm/aws-operator/client/aws"
+	"github.com/giantswarm/aws-operator/service/awsconfig/v7"
 )
 
 type DrainerFrameworkConfig struct {
+	G8sClient    versioned.Interface
+	K8sClient    kubernetes.Interface
+	K8sExtClient apiextensionsclient.Interface
+	Logger       micrologger.Logger
+
+	AWS                DrainerFrameworkConfigAWS
+	GuestUpdateEnabled bool
+	ProjectName        string
+}
+
+type DrainerFrameworkConfigAWS struct {
+	AccessKeyID     string
+	AccessKeySecret string
+	Region          string
+	SessionToken    string
 }
 
 func NewDrainerFramework(config DrainerFrameworkConfig) (*framework.Framework, error) {
-	return nil, nil
+	if config.G8sClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.G8sClient must not be empty", config)
+	}
+	if config.K8sClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.K8sClient must not be empty", config)
+	}
+	if config.K8sExtClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.K8sExtClient must not be empty", config)
+	}
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+
+	if config.AWS.AccessKeyID == "" {
+		return nil, microerror.Maskf(invalidConfigError, "%T.AWS.AccessKeyID must not be empty", config)
+	}
+	if config.AWS.AccessKeySecret == "" {
+		return nil, microerror.Maskf(invalidConfigError, "%T.AWS.AccessKeySecret must not be empty", config)
+	}
+	if config.AWS.Region == "" {
+		return nil, microerror.Maskf(invalidConfigError, "%T.AWS.Region must not be empty", config)
+	}
+	if config.ProjectName == "" {
+		return nil, microerror.Maskf(invalidConfigError, "%T.ProjectName must not be empty", config)
+	}
+
+	var err error
+
+	var crdClient *k8scrdclient.CRDClient
+	{
+		c := k8scrdclient.DefaultConfig()
+
+		c.K8sExtClient = config.K8sExtClient
+		c.Logger = config.Logger
+
+		crdClient, err = k8scrdclient.New(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	resourceRouter, err := newDrainerResourceRouter(config)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	var newInformer *informer.Informer
+	{
+		c := informer.DefaultConfig()
+
+		c.Watcher = config.G8sClient.ProviderV1alpha1().AWSConfigs("")
+
+		c.ResyncPeriod = 30 * time.Second
```

Comment:
The lifecycle resource will be reconciled every 30 seconds. This number is based on a gut feeling. We can adapt later if necessary.

Reply:
That makes sense as a starting point. It can be tuned later.